apigee_edge.module

```diff
@@ -1267,6 +1267,36 @@ function template_preprocess_app_credential(array &$variables) {
   }
 }
 
+/**
+ * Forcing user email address to lowercase for ApigeeX/Hybrid Org.
+ * @see https://github.com/apigee/apigee-edge-drupal/issues/730
+ */
+function apigee_edge_user_email_check(&$account) {
+  $email = $account->getEmail();
+  try {
+    /** @var \Drupal\apigee_edge\SDKConnectorInterface $sdk_connector */
+    $sdk_connector = \Drupal::service('apigee_edge.sdk_connector');
+    $org_controller = \Drupal::service('apigee_edge.controller.organization');
+    /* @var \Apigee\Edge\Api\Management\Entity\Organization $organization */
+    $organization = $org_controller->load($sdk_connector->getOrganization());
+
+    // Check if org is ApigeeX.
+    if ($organization && ('CLOUD' === $organization->getRuntimeType() || 'HYBRID' === $organization->getRuntimeType())) {
+       // Apigee X Org is only accepting lower case email address,
+       // therefore preventing the form from creating account
+       // with uppercase email address.
+       if (preg_match('/[A-Z]/', $email)) {
+        $account->set('mail', strtolower($email));
+        \Drupal::messenger()->addWarning(t('Apigee X Org is only accepting lowercase email address.'));
```

Comment:
So what...? I mean this message is going to be seen by and end user. It does not say that they should change the format of their email address....it also does not explain that the system auto fixed that...and as an end user I probably should not know that my API provider is using Apigee X.

Reply:
Updated.